tensorflow_lite_support/c/test/task/vision/image_classifier_test.cc

```diff
@@ -0,0 +1,302 @@
+#include "tensorflow_lite_support/c/test/task/vision/utils/image_utils.h"
+#include "tensorflow_lite_support/cc/port/gmock.h"
+#include "tensorflow_lite_support/cc/port/gtest.h"
+#include "tensorflow_lite_support/cc/test/test_utils.h"
+
+namespace tflite {
+namespace task {
+namespace vision {
+namespace {
+
+using ::tflite::task::JoinPath;
+
+constexpr char kTestDataDirectory[] =
+    "tensorflow_lite_support/cc/test/testdata/task/vision/";
+// Quantized model.
+constexpr char kMobileNetQuantizedWithMetadata[] =
+    "mobilenet_v1_0.25_224_quant.tflite";
+
+CImageData LoadImage(const char* image_name) {
+  return CDecodeImageFromFile(
+      JoinPath("./" /*test src dir*/, kTestDataDirectory, image_name).data());
+}
+
+TEST(CImageClassifierFromOptionsTest, FailsWithMissingModelPath) {
+  TfLiteImageClassifierOptions options = {0};
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(&options);
+  ASSERT_EQ(image_classifier, nullptr);
+}
+
+TEST(CImageClassifierFromOptionsTest, SucceedsWithModelPath) {
+  //   TfLiteImageClassifierOptions* options =
+  //   TfLiteImageClassifierOptionsCreate();
+  const char* model_path = JoinPath("./" /*test src dir*/, kTestDataDirectory,
+                                    kMobileNetQuantizedWithMetadata)
+                               .data();
+  TfLiteImageClassifierOptions options = {0};
+  options.base_options.model_file.file_path = model_path;
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(&options);
+  EXPECT_NE(image_classifier, nullptr);
+  TfLiteImageClassifierDelete(image_classifier);
+}
+
+TEST(CImageClassifierFromOptionsTest, SucceedsWithNumberOfThreads) {
+  const char* model_path = JoinPath("./" /*test src dir*/, kTestDataDirectory,
+                                    kMobileNetQuantizedWithMetadata)
+                               .data();
+
+  TfLiteImageClassifierOptions options = {0};
+  options.base_options.model_file.file_path = model_path;
+  options.base_options.compute_settings.num_threads = 3;
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(&options);
+  EXPECT_NE(image_classifier, nullptr);
+  TfLiteImageClassifierDelete(image_classifier);
+}
+
+TEST(CImageClassifierFromOptionsTest,
+     FailsWithClassNameBlackListAndClassNameWhiteList) {
+  const char* model_path = JoinPath("./" /*test src dir*/, kTestDataDirectory,
+                                    kMobileNetQuantizedWithMetadata)
+                               .data();
+
+  TfLiteImageClassifierOptions options = {0};
+  options.base_options.model_file.file_path = model_path;
+
+  char* class_name_blacklist[] = {"brambling"};
+  options.classification_options.class_name_blacklist.list =
+      class_name_blacklist;
+  options.classification_options.class_name_blacklist.length = 1;
+
+  char* class_name_whitelist[] = {"cheeseburger"};
+  options.classification_options.class_name_whitelist.list =
```

Comment:
Forgot to mention, this `white/blacklist` need to be renamed to `allow/denylist`  in our new API. It's OK to fix it in another PR. Please add a todo in image_classifier_options.h, such as:
```
TODO(prianka): change `white/blacklist` to `allow/denylist`.
```

Reply:
Added TODOs